packages/wrk2/index.mjs

```diff
@@ -0,0 +1,167 @@
+  body;
+
+  constructor (opts = {}) {
+    // autocannon options
+    this.duration = opts.duration || this.duration;
+    this.connections = opts.connections || this.connections;
+    this.rate = opts.rate || this.rate;
+
+    // request options
+    this.method = opts.method || this.method;
+    this.url = opts.url || this.url;
+    this.headers = opts.headers || this.headers;
+    this.body = opts.body || this.body;
+  }
+
+  async start (opts = {}) {
+    // Check if present, but only once
+    if (this.isPresent === null) {
+      try {
+        await this.spawn(['-h']);
+        this.isPresent = true;
+      } catch (e) {
+        // Set the status to the error we got
+        this.status = e; 
+        this.isPresent = false;
+      }
+    }
+
+    if (!this.isPresent) {
+      this.status = this.status || 'not present';
+      throw new Error('executable not present', {
+        cause: this.status
+      });
+    }
+
+    this.status = 'starting';
+
+    const args = [
+      '-d', this.duration,
+      '-c', this.connections,
+      '-R', this.rate,
+      '-t', Math.min(this.connections, availableParallelism() || 8),
+      '--latency'
+    ];
+
+    // TODO: I don't see docs on how to do this with wrk2
+    if (opts.method || this.method) {
+      throw new Error('not yet supported');
+    }
+
+    // TODO: I don't see docs on how to do this with wrk2
+    if (opts.body || this.body) {
+      throw new Error('not yet supported');
+    }
+
+    // -H/--headers K=V
+    for (const [header, value] of Object.entries(opts.headers || this.headers)) {
+      args.push('-H', `${header}: ${value}`);
+    }
+
+    // url (positional)
+    const url = opts.url ? new Url(opts.url, this.url) : this.url;
+    args.push(url.toString());
+
+    // Run the process
+    const stdout = await this.spawn(args);
+
+    const result = this.processResults(stdout);
+    if (!result.requests.average) {
+      throw new Error(`${this.executable} produced invalid JSON output`);
+    }
+
+    return result;
+  }
+
+  async spawn (args = [], opts = {}) {
+    return new Promise((resolve, reject) => {
+      const proc = spawn(`${this.executable} ${shellQuote(args)}`, {
+        shell: true
+      }, opts);
+
+      this.procs.push(proc);
+
+      let stderr = '';
+      proc.stderr.setEncoding('utf8');
+      proc.stderr.on('data', (chunk) => {
+        stderr += chunk
+      });
+
+      let stdout = '';
+      proc.stdout.setEncoding('utf8');
+      proc.stdout.on('data', (chunk) => {
+        stdout += chunk
+      });
+
+      const onError = (e, data) => {
+        Object.assign(e, { stderr, stdout, ...data });
+        this.status = e;
+        reject(e);
+      }
+
+      proc.once('error', (e) => onError(e));
+      proc.once('close', (code) => {
+        if (code) {
+          return onError(new Error(`${this.executable} failed with ${code}`), { code });
+        }
+
+        resolve(stdout);
+      });
+
+    });
+  }
+
+  processResults (output) {
```

Comment:
Since I didn't get `wkr2` built locally I also didn't test this after the refactor. I think I faithfully copied it but this needs more testing before we land this.